pandas/io/packers.py

```diff
@@ -319,6 +322,18 @@ def encode(obj):
                     u'dtype': u(obj.dtype.name),
                     u'data': convert(obj.values),
                     u'compress': compressor}
+
+    elif isinstance(obj, Categorical):
+        return {u'typ': u'categorical',
+                u'klass': u(obj.__class__.__name__),
+                u'name': getattr(obj, 'name', None),
+                u'codes_dtype': u(obj._codes.dtype.name),
```

Comment:
I don't think you actually need to specify the ``codes/catgories`` dtypes. Those will be automatic. IOW, don't use ``convert`` here, just same ``codes : obj.codes`` and ``categories : obj.categories``. These are recursive encoder/decoders.